File: frontend/src/types/prediction.ts
export interface PredictionPoint {
  date: string; // "YYYY-MM-DD"
  predicted: number;
  lower: number;
  upper: number;
  is_anomaly: boolean;
}

export interface PredictionLatestResponse {
  run_id: string;
  generated_at: string; // ISO datetime
  training_days: number;
  model_type: string;
  series: Record<string, PredictionPoint[]>;
}

export interface PredictionGenerateResponse {
  task_id: string;
  status: string;
}

export interface TaskStatusResponse {
  task_id: string;
  status: string; // PENDING | STARTED | SUCCESS | FAILURE | RETRY
  result: Record<string, unknown> | null;
}

File: app/schemas/prediction.py
"""Pydantic schemas for prediction endpoints."""

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, field_serializer


class PredictionPoint(BaseModel):
    """Single forecast point for one target date."""

    model_config = ConfigDict(from_attributes=True)

    date: date
    predicted: Decimal
    lower: Decimal
    upper: Decimal
    is_anomaly: bool = False

    @field_serializer("predicted", "lower", "upper")
    def serialize_decimal(self, v: Decimal) -> float:
        return float(v)


class PredictionLatestResponse(BaseModel):
    """Response from GET /predictions/latest."""

    run_id: str
    generated_at: datetime
    training_days: int
    model_type: str
    series: dict[str, list[PredictionPoint]]


class PredictionGenerateResponse(BaseModel):
    """Response from POST /predictions/generate."""

    task_id: str
    status: str


class TaskStatusResponse(BaseModel):
    """Response from GET /predictions/tasks/{task_id}."""

    task_id: str
    status: str
    result: dict | None = None
